#  Exploratory Data Analysis and Data Preprocessing

## Problem statement

####Dataset: Google Store App Rating Data
The Play Store apps data has enormous potential to drive app-making businesses to success. However, many apps are being developed every single day and only a few of them become profitable.  It is important for developers to be able to predict the success of their app and incorporate features which makes an app successful.
We can collect app data and user ratings from the app stores and use it to extract insightful information.
A machine learning model can be used to predict rating for a given app, which can be used to estimate success and scope of improvement.


As part of this exercise we will perform EDA on various features like price, number of reviews, rating, etc.



## Data Set Information

Attributes:

- App: Application name

- Rating: Overall user rating of the app

- Reviews: Number of user reviews for the app

- Type: Paid or Free

- Price: Price of the app

- Content Rating: Age group the app is targeted at - Children / Mature 21+ / Adult

## Learning Objective
1. Statistical analysis of the data
2. Duplicate value identification and treatment
   - Create a copy of the dataframe
   - Dropping duplicated records


## Import required libraries

In [3]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import warnings
warnings.filterwarnings('ignore')

## Load the dataset

In [4]:
#df = pd.read_csv('app.csv')

In [12]:
from google.colab import drive
drive.mount('/content/drive')

df = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/Data_preprocessing/app.csv')
df.head()

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


,App,Rating,Reviews,Type,Price,Content Rating
0,Photo Editor & Candy Camera & Grid & ScrapBook,4.1,159,Free,0,Everyone
1,Coloring book moana,3.9,967,Free,0,Everyone
2,"U Launcher Lite – FREE Live Cool Themes, Hide ...",4.7,87510,Free,0,Everyone
3,Sketch - Draw & Paint,4.5,215644,Free,0,Teen
4,Pixel Draw - Number Art Coloring Book,4.3,967,Free,0,Everyone


# 1. EDA and Preprocessing

##  1.1 Explore the data

In [13]:
# checking top 5 samples
df.head(5)

,App,Rating,Reviews,Type,Price,Content Rating
0,Photo Editor & Candy Camera & Grid & ScrapBook,4.1,159,Free,0,Everyone
1,Coloring book moana,3.9,967,Free,0,Everyone
2,"U Launcher Lite – FREE Live Cool Themes, Hide ...",4.7,87510,Free,0,Everyone
3,Sketch - Draw & Paint,4.5,215644,Free,0,Teen
4,Pixel Draw - Number Art Coloring Book,4.3,967,Free,0,Everyone


In [14]:
# info of the dataset
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9360 entries, 0 to 9359
Data columns (total 6 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             9360 non-null   object 
 1   Rating          9352 non-null   float64
 2   Reviews         9360 non-null   int64  
 3   Type            9344 non-null   object 
 4   Price           9360 non-null   object 
 5   Content Rating  9360 non-null   object 
dtypes: float64(1), int64(1), object(4)
memory usage: 438.9+ KB


- There are 6 columns and 9360 rows
- Except rating and reviews, other features are of object type
- There are some null values

In [15]:
# Checking summary Statistics
df.describe()

,Rating,Reviews
count,9352.000000,9.360000e+03
mean,4.191948,5.143767e+05
std,0.515403,3.145023e+06
min,1.000000,1.000000e+00
25%,4.000000,1.867500e+02
50%,4.300000,5.955000e+03
75%,4.500000,8.162750e+04
max,5.000000,7.815831e+07


In [16]:
df.describe(include='O')

,App,Type,Price,Content Rating
count,9360,9344,9360,9360
unique,8190,2,73,6
top,ROBLOX,Free,0,Everyone
freq,9,8700,8715,7414


### Basic observations
- App : 9360 unique values. Top value seems to ROBOLOX. We could have duplicate entries here as top app has 9 instances.
- Rating - min value is 1 and max is 5
- Reviews - there are some apps with zero reviews
- Type - Most of the apps are free. There are only three categories here.
- Price - More than 8.5k apps are free of cost

### Inferences
1. App name can be dropped after doing basic analysis

2. Type & Content Rating - Convert it to numeric

3. Price - The datatype is object type. We should make it numeric

4. Type and Price could be highly correlated

## 1.2 Duplicate value identification and treatment

#### Let's look closely at the apps in the data (App Column) and drop duplicates if there are any

In [17]:
# Create a copy of the dataframe
df1 = df.copy()

In [18]:
# Let's look closely at the apps in the data
df1['App'].value_counts().head(20)

,count
App,
ROBLOX,9
"CBS Sports App - Scores, News, Stats & Watch Live",8
Duolingo: Learn Languages Free,7
ESPN,7
8 Ball Pool,7
Candy Crush Saga,7
Sniper 3D Gun Shooter: Free Shooting Games - FPS,6
"Bleacher Report: sports news, scores, & highlights",6
Zombie Catchers,6


We see that few apps have more than one entries. Which tells us that there few duplicate entries for these apps.

Before we drop the app column, we will verify if these entries are really duplicate and then remove these.

Take 3 sample Apps for exploration (for eg: 'ROBLOX', 'ESPN','Nick')

In [19]:
#Taking 3 sample Apps for exploration
df1[df1['App'].isin(['ROBLOX', 'ESPN','Nick'])].sort_values(by='App')

,App,Rating,Reviews,Type,Price,Content Rating
3913,ESPN,4.2,521081,Free,0,Everyone 10+
2931,ESPN,4.2,521140,Free,0,Everyone 10+
2919,ESPN,4.2,521140,Free,0,Everyone 10+
2907,ESPN,4.2,521140,Free,0,Everyone 10+
2877,ESPN,4.2,521138,Free,0,Everyone 10+
2869,ESPN,4.2,521138,Free,0,Everyone 10+
2818,ESPN,4.2,521138,Free,0,Everyone 10+
864,Nick,4.2,123279,Free,0,Everyone 10+
2141,Nick,4.2,123322,Free,0,Everyone 10+
5685,Nick,4.2,123309,Free,0,Everyone 10+


In [20]:
# Check duplicates
df1.duplicated().sum()

np.int64(477)

### Drops other duplicate entries keeping the first one

In [21]:
# drops other duplicate entries
df1.drop_duplicates('App',inplace=True)

### Verify that duplicates has been removed

In [22]:
# Check duplicates
df1.duplicated().sum()

np.int64(0)

In [23]:
#Let's verify that duplicates has been removed
df1['App'].value_counts()

,count
App,
iHoroscope - 2018 Daily Horoscope & Astrology,1
Photo Editor & Candy Camera & Grid & ScrapBook,1
Coloring book moana,1
"U Launcher Lite – FREE Live Cool Themes, Hide Apps",1
Sketch - Draw & Paint,1
...,...
ibis Paint X,1
FlipaClip - Cartoon animation,1
350 Diy Room Decor Ideas,1


### Check the updated shape of the dataframe

In [24]:
df1.shape

(8190, 6)

--------------
## Happy Learning :)
--------------